implementations/rust/ockam/ockam_app_lib/src/incoming_services/commands.rs

```diff
@@ -0,0 +1,212 @@
+                    return Ok(Some(bind_address.port()));
+                }
+            }
+        }
+
+        self.create_inlet(background_node_client, service)
+            .await
+            .map(Some)
+    }
+
+    /// Create the tcp-inlet for the accepted invitation
+    /// Returns the inlet SocketAddr
+    async fn create_inlet(
+        &self,
+        background_node_client: Arc<dyn BackgroundNodeClient>,
+        service: &IncomingService,
+    ) -> crate::Result<u16> {
+        debug!(
+            service_name = service.name(),
+            "Creating TCP inlet for accepted invitation"
+        );
+
+        let local_node_name = service.local_node_name()?;
+
+        let bind_address = match service.address() {
+            Some(address) => address,
+            None => get_free_address()?,
+        };
+
+        background_node_client
+            .projects()
+            .enroll(
+                &local_node_name,
+                &service.enrollment_ticket()?.hex_encoded()?,
+            )
+            .await?;
+
+        // Recreate the node using the trust context
+        debug!(node = %local_node_name, "Creating node to host TCP inlet");
+        let _ = self.delete_background_node(&local_node_name).await;
+        background_node_client
+            .nodes()
+            .create(&local_node_name)
+            .await?;
+        tokio::time::sleep(Duration::from_millis(250)).await;
+
+        let mut inlet_node = self.background_node(&local_node_name).await?;
+        inlet_node.set_timeout(Duration::from_secs(5));
+        inlet_node
+            .create_inlet(
+                &self.context(),
+                &bind_address.to_string(),
+                &MultiAddr::from_str(&service.service_route()?).into_diagnostic()?,
+                &Some(service.inlet_name().to_string()),
+                &None,
+                Duration::from_secs(5),
+            )
+            .await?;
+        Ok(bind_address.port())
+    }
+
+    pub(crate) async fn disable_tcp_inlet(&self, invitation_id: &str) -> crate::Result<()> {
+        // first change the in-memory state
+        let service = {
+            let incoming_services_arc = self.incoming_services();
+            let mut writer = incoming_services_arc.write().await;
+            let mut service = writer.find_mut_by_id(invitation_id);
+            if let Some(service) = service.as_mut() {
+                if !service.enabled() {
+                    debug!(node = %service.local_node_name()?, alias = %service.name(), "TCP inlet was already disconnected");
+                    return Ok(());
+                }
+                service.disable();
+            }
+            service.cloned()
+        };
+
+        if let Some(service) = service {
+            // this is an async operation, let's give feedback to the user as soon as possible
+            self.publish_state().await;
+            // change the persistent state
+            self.model_mut(|model| {
```

Comment:
I'm not a great fan of this interface to the model state. Why not have a function like:
```rust
impl ModelState {
    pub async fn disable_service(&mut self, invitation_id: &str) -> Result<()> {
       // and use model_mut internally
    }
}
```
I think this would encapsulate a bit better the modifications that we expect to do on a `ModelState`.

Reply:
Makes sense, but I would like to take a bit more time for this, I've created an issue for it!